torch/nn/modules/module.py

```diff
@@ -366,6 +366,11 @@ def to(self, *args, **kwargs):
 
         """
 
+        if len(args) > 3:
+            raise TypeError("to() takes at most 3 arguments ({} given)".format(len(args)))
+        if "copy" in kwargs:
```

Comment:
This isn't comprehensive since it allows `(dtype, bool, bool)`. Could you make the change in c++ arg parsing code? I'm not sure what the best way to fix is.. but maybe just an `if` statement and allow `copy` in one branch?

Reply:
OK.

Reply:
Thinking about this in more detail: Pushing this down to C++ means to keep track of the use case through THPVariable__parse_to (python_nn_functions.cpp in the autograd templates) and parse_to_conversion (autograd/utils/python_arg_parsing.h), which I think might be excessive.

In the end sharing parse_to is a hack and it would look that it's better to check with the caller that it works, i.e. in Python. For that, one could either amend the positional check or make the copy= argument keyword only and forbid that.

If we don't provide separate signatures, the C++ option would have trouble producing a good error message: When the signature with copy matches, we can say "to does not accept 'copy'", but it'd be an awkward error when it was passed as a positional argument.
I'll do a C++ version, but I'd not be opposed if you prefer the python route after seeing it.